### ISSUE TITLE
save_period?

### ISSUE THREAD

**Issue by sarpx** [ultralytics/ultralytics#22589]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

I started segmentation training in yolov11 and made save_period=50. The weights in 250 cycles are 250mb while the weights in 300 cycles are 125mb. Is this normal?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @sarpx, thank you for your interest in Ultralytics 🚀! This is an automated response to help get things moving quickly—an Ultralytics engineer will also review and assist you here soon.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question (e.g., about segmentation checkpoints or `save_period`), please share:
-Exact command or Python snippet you used (including `save_period`), model variant (e.g., YOLO11-seg), and any custom YAMLs
-Environment details: `ultralytics` version, PyTorch version, OS, GPU/CPU, CUDA
-Relevant training logs around the epochs of interest (e.g., 250 and 300) and the exact checkpoint files with their sizes
-Whether you used `resume=True`, DDP, or custom callbacks
-Screenshots/snippets of your runs directory structure and `save_dir`
And verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a Python>=3.8 environment with PyTorch>=1.8 to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including CUDA/CUDNN, Python and PyTorch preinstalled):
-Notebooks with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
-Google Cloud Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
-Amazon Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
-Docker Image. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Once you share the details above, we’ll help pinpoint what’s happening with your segmentation checkpoints and `save_period` behavior 🔍💾

**Comment by Y-T-G** [comment]
The final weights are in FP16, while intermediate weights are in FP32.

**Comment by sarpx** [comment]
@Y-T-G  I'm trying to get the most accurate generalization with the inner setting, and it's easier to get this in an intermediate epoch, but it's very difficult to get the outputs to be twice the size of the other one. Isn't it possible to save the intermediate weights with 16?

**Comment by Y-T-G** [comment]
You can reduce the size manually 

https://www.reddit.com/r/Ultralytics/comments/1hdwx7d/reducing_the_size_of_the_weights_after/
